agent/api/task/task_attachment_handler.go

```diff
@@ -0,0 +1,114 @@
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package task
+
+import (
+	"fmt"
+
+	"github.com/aws/amazon-ecs-agent/agent/acs/model/ecsacs"
+	"github.com/aws/amazon-ecs-agent/agent/logger"
+	"github.com/aws/aws-sdk-go/aws"
+)
+
+const (
+	// serviceConnectAttachmentType specifies attachment type for service connect
+	serviceConnectAttachmentType = "ServiceConnect"
+)
+
+// AttachmentHandler defines an interface to handel attachment received from ACS.
+type AttachmentHandler interface {
+	parseAttachment(acsAttachment *ecsacs.Attachment) error
+	validateAttachment(acsTask *ecsacs.Task) error
+	getHandlerType() string
+}
+
+// ServiceConnectAttachmentHandler defines a service connect type attachment handler.
+type ServiceConnectAttachmentHandler struct {
+	scConfig *ServiceConnectConfig
+}
+
+// NewAttachmentHandlers returns all type of handlers to handle different type of attachment.
+func NewAttachmentHandlers() []AttachmentHandler {
+	scAttachmentHandler := &ServiceConnectAttachmentHandler{}
+	handlers := []AttachmentHandler{scAttachmentHandler}
+	return handlers
+}
+
+// attachment parser of service connect attachment handler.
+func (scAttachment *ServiceConnectAttachmentHandler) parseAttachment(acsAttachment *ecsacs.Attachment) error {
+	config, err := ParseServiceConnectAttachment(acsAttachment)
+	scAttachment.scConfig = config
+	return err
+}
+
+// attachment validator of service connect attachment handler.
+func (scAttachment *ServiceConnectAttachmentHandler) validateAttachment(acsTask *ecsacs.Task) error {
+	config := scAttachment.scConfig
+	taskContainers := acsTask.Containers
+	networkMode := aws.StringValue(acsTask.NetworkMode)
+	ipv6Enabled := false
+	if acsTask.ElasticNetworkInterfaces != nil {
+		for _, eni := range acsTask.ElasticNetworkInterfaces {
+			if len(eni.Ipv6Addresses) != 0 {
+				ipv6Enabled = true
+				break
+			}
+		}
+	}
+	return ValidateServiceConnectConfig(config, taskContainers, networkMode, ipv6Enabled)
+}
+
+// attachment handler type of service connect attachment handler.
+func (scAttachment *ServiceConnectAttachmentHandler) getHandlerType() string {
+	return serviceConnectAttachmentType
+}
+
+// HandleTaskAttachments parses and validates attachments based on attachment type.
+func HandleTaskAttachments(acsTask *ecsacs.Task) (*ServiceConnectConfig, error) {
+	if acsTask.Attachments != nil {
+		attachmentHandlers := NewAttachmentHandlers()
+		var serviceConnectAttachment *ecsacs.Attachment
+		var serviceConnectConfig *ServiceConnectConfig
+
+		for _, attachment := range acsTask.Attachments {
+			switch aws.StringValue(attachment.AttachmentType) {
+			case serviceConnectAttachmentType:
+				serviceConnectAttachment = attachment
+			default:
+				logger.Debug("Received an attachment type", logger.Fields{
+					"attachmentType": attachment.AttachmentType,
+				})
+			}
+		}
+
+		for _, h := range attachmentHandlers {
```

Comment:
Instead of iterating attachments and handlers separately, could we add another method to return the handler given the type of Attachment.
```
func getHandler(AttachmentType) Handler 
``` 
That way we could simplify the overall attachment processing when we add more types.

Reply:
Thanks for the suggestion! A new function `getHandlerByType` is added and used in `HandleTaskAttachments`. Please see the update in commit 038dca8ba6b3da87074340d9ac535870a2ca39a6. Thanks.